### PR TITLE
381 remove slices_field function

### PR DESCRIPTION
**Pull request by kartikeyakirar**
this PR fixes https://github.com/insightsengineering/teal.slice/issues/381

### CODE REVIEW

File: R/FilterStates.R [hunk 1]
```diff
@@ -271,7 +271,7 @@ FilterStates <- R6::R6Class( # nolint
         }
 
         # Drop teal_slices that refer to excluded variables.
-        varnames <- slices_field(state, "varname")
+        varnames <- unique(unlist(lapply(state, function(x) x[["varname"]])))
```
Comment (gogonzo) [inline R/FilterStates.R:274]: FYI @chlebowa I think lapply is fine as not all `teal_slice` objects have `"varname"`. This means we can't use `vapply` in all cases

File: R/FilterStates.R [hunk 2]
```diff
@@ -438,7 +438,7 @@ FilterStates <- R6::R6Class( # nolint
           avail_column_choices <- reactive({
             data <- private$data
             vars_include <- private$get_filterable_varnames()
-            active_filter_vars <- slices_field(self$get_filter_state(), "varname")
+            active_filter_vars <- unique(unlist(lapply(self$get_filter_state(), function(x) x[["varname"]])))
```
Comment (gogonzo) [inline R/FilterStates.R:441]: shorter version
```suggestion
            active_filter_vars <- unique(unlist(lapply(self$get_filter_state(), `[[`, "varname")))
```

File: R/FilteredData.R [hunk 3]
```diff
@@ -486,7 +486,7 @@ FilteredData <- R6::R6Class( # nolint
         }
 
         checkmate::assert_class(state, "teal_slices")
-        datanames <- slices_field(state, "dataname")
+        datanames <- unique(vapply(state, "[[", character(1L), "dataname"))
```
Comment (gogonzo) [inline R/FilteredData.R:489]: Here we can use `vapply` as all `teal_slice` objects have "dataname" 👍

File: R/FilterStates.R (hunk 1 above)
Comment (chlebowa) [reply R/FilterStates.R:274]: [My point exactly.](https://github.com/insightsengineering/teal.slice/issues/381#issuecomment-1640176164)

**Comment by kartikeyakirar** [comment]
> 👍 I'd prefer `lapply(, '[[', "field")` instead of `lapply(, function(x) x[["field"]])`. But PR fixes the "issue"

I have updated in all the calls.

**Pull request merged**